iModelCore/ecobjects/PublicApi/ECObjects/ECSchema.h

```diff
@@ -2929,6 +2929,19 @@ struct SearchPathSchemaFileLocater : IECSchemaLocater, RefCountedBase, NonCopyab
     ECOBJECTS_EXPORT static SearchPathSchemaFileLocaterPtr CreateSearchPathSchemaFileLocater(bvector<WString> const& searchPaths, bool includeFilesWithNoVerExt=false);
 };
 
+//=======================================================================================
+//! Reads schemas from a given set of Xml strings
+//=======================================================================================
+struct EXPORT_VTABLE_ATTRIBUTE StringSchemaLocater : IECSchemaLocater, NonCopyableClass
+{
+private:
+    bmap<Utf8String, Utf8String> m_schemaXml;
+protected:
+    ECOBJECTS_EXPORT ECSchemaPtr _LocateSchema(SchemaKeyR key, SchemaMatchType matchType, ECSchemaReadContextR schemaContext) override;
+public:
+    ECOBJECTS_EXPORT void AddSchemaString(Utf8String schemaName, Utf8String schemaXml) {m_schemaXml[schemaName] = schemaXml;}

```

Comment:
Using Utf8String here is problematic. It means the value gets passed and stored by value, which means the huge schema string gets copied multiple times.
We should probably store this as Utf8String anyways because the caller may delete the string before our locater gets to reading it, or we may make that a documented limitation of the StringSchemaLocater.
In any case, you should change the parameters of this method into either Utf8StringR or Utf8CP

Reply:
Updated parameters of AddSchemaString() method as SchemaKeyCR and Utf8StringCR